meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -0,0 +1,294 @@
+var current_Year = (new Date()).getFullYear();
+
+function gatherTabletData(locID) {
+
+    // First destroyany pre-existing table.
+    $('#div_tabletCompletness table').bootstrapTable('destroy');
+    $('#div_tabletCompletness table').remove();
+
+    //Clear the arrays
+    columnNameArray = [];
+    tabletDataArray = [];
+
+    //API URLS to get Data
+    buildTableStructure();
+    var locationAPI = "/locations";
+    var week2 = parseInt(week) - 2;
+    var week1 = parseInt(week) - 1;
+
+    var main_week_tot = '/devices/submissions/tot_1?location=' + locID + '&filter=epi_year:eq:' + current_Year + '&filter=epi_week:eq:';
```

Comment:
Just pondering whether these API urls would look neater stored in a single JS/JSON object rather than so many new variables.  I think it would be more readable personally, but there is not much in it so it is your call.

Reply:
Done, built an array to handle the APIs